### PR TITLE
LibWeb: Implement the default sizing algorithm steps for backgrounds

### PR DESCRIPTION
**Pull request by MacDue**
| Before                                                                                                | After                                                                                                 |
|-------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------|
| image | image |

Fixes https://github.com/SerenityOS/serenity/issues/20992

### CODE REVIEW

File: Tests/LibWeb/Ref/reference/svg-background-no-natural-size-ref.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  body {
+    background-color: white;
+  }
+</style>
+<!-- To rebase:
+  1. Open svg-background-no-natural-size.html in Ladybird
+  2. Right click > "Take Full Screenshot"
+  3. Update the image below:
+-->
+<img src="./images/svg-background-no-natural-size-ref.png">
```
Comment (nico) [inline Tests/LibWeb/Ref/reference/svg-background-no-natural-size-ref.html:14]: Isn't Ref for ref tests, while this is basically a pixel test?

Comment (MacDue) [reply Tests/LibWeb/Ref/reference/svg-background-no-natural-size-ref.html:14]: There's a bunch of tests like this already (I don't think there's a separate place for pixel tests)

Comment (awesomekling) [reply Tests/LibWeb/Ref/reference/svg-background-no-natural-size-ref.html:14]: Right, we don't have a dedicated mechanism for pixel tests atm.

**Comment by MacDue** [comment]
The test is flakey as the screenshot for the ref-test can be (randomly) taken before the background image loads, not sure how to avoid that other than somehow adding in a delay.

**Comment by shannonbooth** [comment]
> The test is flakey as the screenshot for the ref-test can be (randomly) taken before the background image loads, not sure how to avoid that other than somehow adding in a delay.

Sounds like https://github.com/SerenityOS/serenity/issues/21957

**Comment by MacDue** [comment]
> > The test is flakey as the screenshot for the ref-test can be (randomly) taken before the background image loads, not sure how to avoid that other than somehow adding in a delay.
> 
> Sounds like #21957

Yep, looks like it :+1: I saw the hack used in another test and it seems to do the trick for now.

**Pull request merged**